ast/desugar/Desugar.cc

```diff
@@ -317,8 +317,15 @@ ExpressionPtr buildMethod(DesugarContext dctx, core::LocOffsets loc, core::LocOf
     auto enclosingBlockArg = blockArg2Name(dctx, *blkArg);
 
     DesugarContext dctx2(dctx1.ctx, dctx1.uniqueCounter, enclosingBlockArg, declLoc, name);
-    ExpressionPtr desugaredBody = desugarBody(dctx2, loc, body, std::move(destructures));
-    desugaredBody = validateRBIBody(dctx2, move(desugaredBody));
+
+    bool removeBody = dctx.ctx.state.autogenPrintingSubclassesOrAutoloaderOnly;
+    ExpressionPtr desugaredBody;
+    if (!removeBody) {
+        desugaredBody = desugarBody(dctx2, loc, body, std::move(destructures));
+        desugaredBody = validateRBIBody(dctx2, move(desugaredBody));
+    } else {
+        desugaredBody = MK::EmptyTree();
```

Comment:
Can you add a brief comment here to the effect of this being for performance, not correctness?